### PR TITLE
Provide better feedback when gitpod.yml is invalid

### LINKED ISSUE #4828: provide better feedback if .gitpod.yml cannot be parsed
Right now we ignore it completely and does not tell the user the reason.

See https://github.com/gitpod-io/gitpod/issues/4825#issuecomment-880332288 and https://community.gitpod.io/t/gitpod-yml-not-running-init-command-tasks-and-new-images-also-not-building/4098

### PR DESCRIPTION
**Pull request by csweichel**
fixes #4828

This PR adds a proper failure mode for invalid gitpod.yml configuration. Instead of silently falling back to the default config, it surfaces the error and offers the choice to start with the default config.

### How to test
1. https://csweichel-provide-better-feedback-4828.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod-test-repo/tree/broken-gitpod-config

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4844#" title="Author self-approved">csweichel</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **alextugarev** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @alextugarev` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/workspace/config-provider.ts
Comment (geropl) [inline components/server/src/workspace/config-provider.ts:47]: nit: This can be moved into the `try`

File: components/server/src/workspace/config-provider.ts
Comment (geropl) [inline components/server/src/workspace/config-provider.ts:128]: :+1:

File: components/server/src/workspace/config-provider.ts
Comment (geropl) [inline components/server/src/workspace/config-provider.ts:167]: :+1:

(it's a bit hard to parse the diff, but I understood that these three changes are the only relevant ones, the rest seems to be a move)

File: components/server/src/workspace/config-provider.ts
Comment (geropl) [inline components/server/src/workspace/config-provider.ts:51]: :+1:

File: components/server/src/workspace/config-provider.ts
Comment (geropl) [inline components/server/src/workspace/config-provider.ts:348]: We have quite some classes like this around. Would be awesome if we found a way to generalize this approach! :+1:

**Review by geropl — APPROVED** [review]
Code looks good and tested the validation detection: :heavy_check_mark:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 7d683b691369002e0e2b93e724ca7523ce21e6f3</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4844#" title="Author self-approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4844#pullrequestreview-709583746" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [geropl]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/workspace/config-provider.ts
Comment (csweichel) [reply components/server/src/workspace/config-provider.ts:167]: Indeed. Also note the demotion of log prio. If this fails here it's a user input issue, not a system error

**Pull request merged**
